### PR TITLE
Migrate from AAD Pod Identity to Workload Identity (Supported by User Managed Identities) & Update AKS Version

### PR DESCRIPTION
**Pull request by ckittel**
* Replaced Pod Identity with Workload Identity (with Managed Identity)
  * Moved az cli requirement to the version that supports workload _managed_ identity (if a user needs to manipulate via cli)
  * Removed the manifest files related to pod identity
  * Adjust Azure Policy assignments based on the removal of the pod identity pods
  * Remove the managed identity operator role assignment (yay!)
* Also upgraded to AKS 1.24.6 (not related to this change, 1.24.0 is no longer supported)
* Remove the extension manager feature flag (not related to this change, just isn't needed anymore)
* Updated the azure policy assignment names (not related to this change, just reflecting new output)
* Updated some resource references to strong references to avoid bicep warnings on newer versions of the bicep linter (QOL)

This was tested end-to-end. 🚢

### CODE REVIEW

File: 04-networking.md [hunk 1]
```diff
@@ -85,7 +85,7 @@ The following two resource groups will be created and populated with networking
    The spoke creation will emit the following:
 
      * `appGwPublicIpAddress` - The Public IP address of the Azure Application Gateway (WAF) that will receive traffic for your workload.
-     * `clusterVnetResourceId` - The resource ID of the VNet that the cluster will land in. E.g. `/subscriptions/[id]/resourceGroups/rg-enterprise-networking-spokes/providers/Microsoft.Network/virtualNetworks/vnet-spoke-BU0001A0008-00`
+     * `clusterVnetResourceId` - The resource ID of the Virtual network that the cluster, App Gateway, and related will be deployed in. E.g. `/subscriptions/[id]/resourceGroups/rg-enterprise-networking-spokes/providers/Microsoft.Network/virtualNetworks/vnet-spoke-BU0001A0008-00`
```
Comment (hallihan) [inline 04-networking.md:88]:
```suggestion
     * `clusterVnetResourceId` - The resource ID of the Virtual network where the cluster, App Gateway, and related will be deployed. E.g. `/subscriptions/[id]/resourceGroups/rg-enterprise-networking-spokes/providers/Microsoft.Network/virtualNetworks/vnet-spoke-BU0001A0008-00`
```

File: cluster-stamp.bicep [hunk 2]
```diff
@@ -1558,7 +1546,7 @@ resource kvPodMiIngressControllerSecretsUserRole_roleAssignment 'Microsoft.Autho
   }
 }
 
-// Grant the AKS cluster ingress controller pod managed identity with key vault reader role permissions; this allows our ingress controller to pull certificates
+// Grant the AKS cluster ingress controller managed workload identity with key vault reader role permissions; this allows our ingress controller to pull certificates
```
Comment (hallihan) [inline cluster-stamp.bicep:1549]: For consistency with above:
```suggestion
// Grant the AKS cluster ingress controller's managed workload identity with key vault reader role permissions; this allows our ingress controller to pull certificates.
```

**Review by hallihan — APPROVED** [review]
Looks good, just a couple of suggestions

File: 04-networking.md (hunk 1 above)
Comment (RobBagby) [inline 04-networking.md:88]: Are you missing a noun after related? The related what will be deployed?

**Review by RobBagby — APPROVED** [review]

File: 04-networking.md (hunk 1 above)
Comment (ckittel) [reply 04-networking.md:88]: Good reword, update will be coming shortly.

Comment (ckittel) [reply 04-networking.md:88]: Good catch, update will be coming shortly.

File: cluster-stamp.bicep (hunk 2 above)
Comment (ckittel) [reply cluster-stamp.bicep:1549]: Good catch

File: 04-networking.md (hunk 1 above)
Comment (ckittel) [reply 04-networking.md]: Resolved in: https://github.com/mspnp/aks-baseline/pull/326/commits/3f3058684316dffe84b7b6180ccbf5a212c27a94

File: cluster-stamp.bicep (hunk 2 above)
Comment (ckittel) [reply cluster-stamp.bicep]: Resolved in: https://github.com/mspnp/aks-baseline/pull/326/commits/3f3058684316dffe84b7b6180ccbf5a212c27a94

File: 04-networking.md (hunk 1 above)
Comment (ckittel) [reply 04-networking.md]: Resolved in: https://github.com/mspnp/aks-baseline/pull/326/commits/3f3058684316dffe84b7b6180ccbf5a212c27a94

**Comment by ckittel** [comment]
Thanks all for the review.  If there is anything else, just let me know and I'll catch it in a follow-up PR.

**Pull request merged**